route/emitter.go

```diff
@@ -69,15 +71,23 @@ func (e *Emitter) emit(batch []*eirini.Routes) {
 }
 
 func (e *Emitter) unregisterRoute(route *eirini.Routes) {
-	err := e.publish(unregisterSubject, route.UnregisteredRoutes, route.Name)
+	err := e.publish(unregisterSubject, route.UnregisteredRoutes, route.Name, route.ServiceAddress, route.ServicePort, route.ServiceTLSPort)
 	if err != nil {
 		fmt.Println("failed to publish unregistered route:", err.Error())
 	}
 }
 
-func (e *Emitter) publish(subject string, routes []string, name string) error {
+func (e *Emitter) publish(subject string, routes []string, name, host string, httpPort, tlsPort uint32) error {
+
+	// If we use an ingress, all emitted routes should point to it
+	if e.useIngress {
+		host = e.kubeEndpoint
+		httpPort = ingressHTTPPort
+		tlsPort = ingressTLSPort
```

Comment:
We think that this logic should not be in the `route.Emitter`. The Emitter is responsible only for emitting the information it receives and should not care about scheduler specifics (like ingress).

We suggest moving this logic in the `ServiceRouteLister`.

cc @mnitchev